tests/test_c_api.cc

```diff
@@ -0,0 +1,176 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation
+ * nor does it submit to any jurisdiction.
+ */
+
+/// @file   test_c_api.cc
+/// @date   Dec 2024
+/// @author Christopher Bradley
+
+#include "metkit/api/metkit_c.h"
+#include "eckit/testing/Test.h"
+#include "eckit/types/Date.h"
+#include <cstring>
+
+using namespace eckit::testing;
+namespace metkit::test {
+
+// Wrapper around the C function calls that will throw an exception if the function fails
+// i.e. if the return value is not METKIT_SUCCESS
+#define METKIT_TEST_C(expr) \
+    do { \
+        int err = expr; \
+        if (err != METKIT_SUCCESS) { \
+            throw TestException("C-API error: " + std::string(metkit_get_error_string(err)), Here()); \
+        } \
+    } while (false)
+
+// Wrapper around EXPECT(strcmp(a, b) == 0) that will throw an exception if the comparison fails
+#define EXPECT_STR_EQUAL(a, b) \
+    do { \
+        if (strcmp(a, b) != 0) { \
+            throw TestException("Expected: " + std::string(a) + " == " + std::string(b), Here()); \
+        } \
+    } while (false)
+
+// Fairly minimal test coverage
+CASE( "metkit_marsrequest" ) {
```

Comment:
This test is quite long. Could you break it in individual tests for easier and faster comprehension?